pkg/services/app/add.go

```diff
@@ -132,6 +132,23 @@ func (a *App) Add(params AddParams) error {
 	}
 }
 
+func getAppHash(params AddParams) (string, error) {
+	var appHash string
+	var err error
+	if DeploymentType(params.DeploymentType) == DeployTypeHelm {
+		appHash, err = utils.GetAppHash(params.Url, params.Chart, params.Branch)
```

Comment:
Since this `utils.GetAppHash` function accepts three strings, it doesn't really do anything that is application-specific. I can use any three strings and get a valid return value.

Can we change this hashing function to accept `AddParams` and return a hash? That way, to the user of this function it reads as "if you give me the same `AddParams`, I will give you the same content hash every time` (with the caveat that we only care about certain fields).

Reply:
I like that

Reply:
I changed the function `utils.GetAppHash(url, path, branch string)` to be `utils.GetHash(inputs ...string)`(A bit more open as it will now accept N amount of strings). BTW there is a function `getAppHash` already that receives AddParams which I believe fits with the quoated statements you mention.